### PR TITLE
Allow looping within options menu via keyboard

### PR DESCRIPTION
**Pull request by bereket-WMDE**
[T268771](https://phabricator.wikimedia.org/T268771)

### CODE REVIEW

File: vue-components/src/components/OptionsMenu.vue [hunk 1]
```diff
@@ -91,13 +105,26 @@ export default Vue.extend( {
...
+					this.keyboardScroll();
 					break;
 				case 'ArrowDown':
-					this.keyboardHoveredItemIndex = Math.min(
+					if ( this.allowLooping && this.keyboardHoveredItemIndex == this.menuItems.length - 1 ) {
+						// go back to the top of the menu
+						this.keyboardHoveredItemIndex = 0;
+					} else {
+						this.keyboardHoveredItemIndex = Math.min(
 						this.menuItems.length - 1,
```
Comment (micgro42) [inline vue-components/src/components/OptionsMenu.vue:123]: I would have expected the linter to complain about the missing indentation :thinking: 

cc @itamargiv

File: vue-components/src/components/OptionsMenu.vue [hunk 2]
```diff
@@ -106,6 +133,17 @@ export default Vue.extend( {
 					break;
 			}
 		},
+		keyboardScroll(): void {
+			const element = this.$refs[ 'menu-items' ] as HTMLElement[];
+
+			if ( this.scrollWithKeyboard && this.keyboardHoveredItemIndex != -1 ) {
+				element[ this.keyboardHoveredItemIndex ].scrollIntoView( {
```
Comment (micgro42) [inline vue-components/src/components/OptionsMenu.vue:140]: And once you have removed the prop, I think you can fix the tests with this approach: https://stackoverflow.com/a/52446309/3293343

File: vue-components/src/components/OptionsMenu.vue (hunk 1 above)
Comment (itamargiv) [reply vue-components/src/components/OptionsMenu.vue:123]: :+1: I don't think there's a particular rule currently set for that. I would suggest opening a separate task and PR to implement this rule (the "CallExpression" option on the ["indent"](https://eslint.org/docs/rules/indent) rule) and in the meanwhile just suggest the preferred indentation.

**Review by itamargiv — COMMENTED** [review]
Looks good on my side. Thank you @bereket-WMDE, master of keyboard navigation :wink:. Just address the failing tests and comment @micgro42 made, and I'll leave the final approval to him.

Mannaquin head robot rolling on keyboard

**Comment by Ladsgroup** [comment]
> Looks good on my side. Thank you @bereket-WMDE, master of keyboard navigation wink. Just address the failing tests and comment @micgro42 made, and I'll leave the final approval to him.

Michael is in Bahamas until next week.

**Comment by itamargiv** [comment]
> Michael is in Bahamas until next week.

Thanks for letting me know

**Comment by SaiSan-WMDE** [comment]
I just checked, and it appears that the focus disappears first instead of moving directly to the next option (either the first or the last) whenever the down/up arrow keys are pressed. In other words: two keystrokes are needed in order to loop to the first or last options in the menu.

**Comment by bereket-WMDE** [comment]
> Looks good on my side. Thank you @bereket-WMDE, master of keyboard navigation wink. Just address the failing tests and comment @micgro42 made, and I'll leave the final approval to him.
> 
> Mannaquin head robot rolling on keyboard

haha.. thanks. thanks. 
and thank you for looking into it.

**Comment by bereket-WMDE** [comment]
> I just checked, and it appears that the focus disappears first instead of moving directly to the next option (either the first or the last) whenever the down/up arrow keys are pressed. In other words: two keystrokes are needed in order to loop to the first or last options in the menu.

@SaiSan-WMDE can you check again ? it just finished deploying the new version

**Comment by SaiSan-WMDE** [comment]
> > I just checked, and it appears that the focus disappears first instead of moving directly to the next option (either the first or the last) whenever the down/up arrow keys are pressed. In other words: two keystrokes are needed in order to loop to the first or last options in the menu.
> 
> @SaiSan-WMDE can you check again ? it just finished deploying the new version

Looking good! I'm realizing now that I failed to understand the issue. It actually had to do with scroll not being activated via keyboard on Chrome (the last option remains hidden), which I guess will be tackled separately (([T267829](https://phabricator.wikimedia.org/T267829))). Sorry for the silly confusion 🤦

**Comment by bereket-WMDE** [comment]
> > > I just checked, and it appears that the focus disappears first instead of moving directly to the next option (either the first or the last) whenever the down/up arrow keys are pressed. In other words: two keystrokes are needed in order to loop to the first or last options in the menu.
> > 
> > 
> > @SaiSan-WMDE can you check again ? it just finished deploying the new version
> 
> Looking good! I'm realizing now that I failed to understand the issue. It actually had to do with scroll not being activated via keyboard (I was confused because the last option remains hidden in Chrome – while Firefox it all works just fine). I guess thihs will be tackled separately ([T267829](https://phabricator.wikimedia.org/T267829)), so all good. Sorry for the silly confusion facepalm

@SaiSan-WMDE no worries at all. yes, the issue you mentioned is fixed in https://github.com/wmde/wikit/pull/360
thank you for looking into it.

P.S. you can also test that functionality.

**Pull request merged**